.github/workflows/pre-submit.actions.yml

```diff
@@ -299,10 +299,14 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0
+      - name: Detect the builder ref
+        id: detect
+        uses: ./.github/actions/detect-workflow
       - uses: ./.github/actions/generate-builder
         with:
-          repository: "slsa-framework/slsa-github-generator"
-          ref: "refs/tags/v1.2.2"
+          repository: ${{ steps.detect.output.repository }}
+          ref: ${{ steps.detect.outputs.ref }}
+          builder-ref: "refs/tags/v1.2.2"
```

Comment:
should the value of `builder-ref` be retrieved dynamically? Example, the latest version or is does not really matter?
Can you add a comment about it?

Reply:
> should the value of builder-ref be retrieved dynamically?

For users outside of this, we should not be retrieving dynamically. That will cause problems like the cosign-installer problems.

It should always be the detected workflow reference.

Reply:
I added a small comment about avoiding fetching dynamically and that it should always match the reusable workflow reference for callers outside of presubmit.

Reply:
Thanks.

Reply:
This was always hard-coded - we don't need to update it. We are testing that any builder version can be fetched from the releases and verified.

There is no canonical builder version to bump here.

Reply:
We could invoke via a matrix if you wanted to test all? but one was the only path we were testing.

Reply:
sorry, I got confused. I checked the code and it makes sense now. The only reason we're setting`ref: ${{ steps.detect.outputs.ref }}` is because it's a required field, but it's not actually used